sdk/core/core-amqp/test/retry.spec.ts

```diff
@@ -188,24 +188,24 @@ describe("retry function", function() {
         },
         connectionId: "connection-1",
         operationType: RetryOperationType.session,
-        times: 4,
+        maxRetries: 4,
         delayInSeconds: 0.5
       };
       await retry(config);
     } catch (err) {
       should.exist(err);
       should.equal(true, err instanceof MessagingError);
       err.message.should.equal("I would always like to fail, keep retrying.");
-      counter.should.equal(4);
+      counter.should.equal(5);
```

Comment:
How is that all the other `counter.should.equal` checks dont need updates?

Reply:
So the way the tests were written, the error condition happens at attempt # 3 which is what the condition tests, it doesn't matter if more retry attempts were allowed as the non-retryable error gets thrown and handled at count 3. 

But nevertheless, updated tests though not required..